lib/src/analysis/analyzer.rs

```diff
@@ -298,6 +298,35 @@ impl<'de> Deserialize<'de> for AnalysisRow {
     }
 }
 
+#[derive(Serialize, Clone)]
+pub struct DetectionRow {
+    pub packet_timestamp: DateTime<FixedOffset>,
+    pub events: Vec<Event>,
+    pub skipped_message_reason: Option<String>,
+}
+
+impl TryFrom<AnalysisRow> for DetectionRow {
+    type Error = &'static str;
+
+    fn try_from(ar: AnalysisRow) -> Result<DetectionRow, Self::Error> {
+        let events: Vec<Event> = ar.events.into_iter().filter_map(|e| e).collect();
+
+        if events.is_empty() {
```

Comment:
This will break skip reason generation. TryFrom might be a poor fit here because we need to be able to return an output row with n events or a skip reason and TryFrom doesn't make this configurable without passing a tuple, and at that point it's starting to abuse the TryFrom semantics.